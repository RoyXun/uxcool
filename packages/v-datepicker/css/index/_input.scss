@mixin input() {
  position: relative;
  display: inline-block;
  width: 100%;
  height: 25px;
  margin: 0 0;
  padding: 4px 10px;
  border-radius: 6px 6px;
  border: 1px solid #d9d9d9;
  background-color: #ffffff;
  color: #666;
  line-height: 1.5;
  transform: border 0.3s cubic-bezier(0.35, 0, 0.25, 1),
    background 0.3s cubic-bezier(0.35, 0, 0.25, 1), box-shadow 0.3s cubic-bezier(0.35, 0, 0.25, 1);
  &:disabled {
    background: #eee;
  }
}
