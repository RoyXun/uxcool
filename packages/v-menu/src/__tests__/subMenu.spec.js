import {
  mount,
  waitTime,
  getPortal,
  triggerEvent,
  createWrapper,
  mountPickerFactory,
  shallowMount,
} from '@suning/v-test-utils';
import { VMenu, VMenuItem, VSubMenu } from '../index';
import {
  getRootSubMenu, getPopupPlacement, isTopSubMenu, getTitle
} from '../utils';

describe('subMenu', () => {
  it('mouseenter , mouseleave', async () => {
    const wrapper = mount({
      render() {
        return (
          <VMenu mode="horizontal">
            <VSubMenu name="vsubmenu4" title="vsubmenu4" class="vsubmenu4">
              <template slot="title">测试4</template>
            </VSubMenu>
          </VMenu>
        );
      },
    });
    await wrapper.vm.$nextTick();
    wrapper.find('.v-menu-submenu-horizontal').trigger('mouseenter');
    await waitTime();
    await waitTime();
    wrapper.find('.v-menu-submenu-horizontal').trigger('mouseleave');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('mouseenter , mouseleave, and destory Actions', async () => {
    const wrapper = mount({
      render() {
        return (
          <VMenu>
            <VSubMenu name="vsubmenu" title="vsubmenu" class="vsubmenu"></VSubMenu>
          </VMenu>
        );
      },
    });
    await wrapper.vm.$nextTick();
    wrapper.find('.v-menu-submenu-title').trigger('mouseenter');
    await wrapper.vm.$nextTick();
    expect(wrapper.find('.vsubmenu').classes('v-menu-submenu-active')).toBeTruthy();
    wrapper.find('.v-menu-submenu-title').trigger('mouseleave');
    await waitTime();
    expect(wrapper.find('.vsubmenu').classes('v-menu-submenu-active')).toBeFalsy();
    wrapper.destroy();
    await waitTime();
    expect(wrapper.exists()).toBe(false);
  });

  it('destroy', async () => {
    const wrapper = mount({
      render() {
        return (
          <VMenu>
            <VSubMenu name="item-1" title="vsubmenu" class="notitle"></VSubMenu>
          </VMenu>
        );
      },
    });
    await waitTime(20);
    wrapper.destroy();
    expect(wrapper.exists()).toBe(false);
  });

  it('test horizontal', async () => {
    const wrapper = mount({
      render() {
        return (
          <VMenu mode="horizontal">
            <VSubMenu name="VSubMenu" title="VSubMenu"></VSubMenu>
          </VMenu>
        );
      },
    });
    // await waitTime(20);
    await wrapper.vm.$nextTick();
    expect(wrapper.find('.v-menu-submenu').classes('v-menu-submenu-horizontal')).toBeTruthy();
    await waitTime(20);
    wrapper.destroy();
  });

  it('open', async () => {
    const wrapper = mount({
      render() {
        return (
          <VMenu>
            <VSubMenu name="open" title="open" class="open">
              <VMenuItem name="sub-menu-1-1" class="sub-menu-1-1">
                VMenuItem-1
              </VMenuItem>
            </VSubMenu>
          </VMenu>
        );
      },
    });
    await waitTime(20);
    wrapper.find('.v-menu-submenu-title').trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.find('.open').classes('v-menu-submenu-open')).toBeTruthy();
    await waitTime(20);
    wrapper.find('.sub-menu-1-1').trigger('click');
    await wrapper.vm.$nextTick();
    // expect(wrapper.vm.$refs.ccc1.isSelected).toBe(true);
    await waitTime(20);
    wrapper.destroy();
  });

  it('no title', async () => {
    const wrapper = mount({
      render() {
        return (
          <VMenu>
            <VSubMenu name="item-1"></VSubMenu>
          </VMenu>
        );
      },
    });
    await waitTime(20);
    wrapper.destroy();
  });

  it('onPopupVisible', async () => {
    const wrapper = mount({
      data() {
        return {
          level: 2,
          rootMode: 'horizontal',
          isInlineMode: false,
        };
      },
      render() {
        const { isInlineMode } = this;
        return (
          <VMenu isInlineMode={isInlineMode}>
            <VSubMenu name="vsubmenu1" title="vsubmenu1" class="vsubmenu1" ref="testPop"></VSubMenu>
          </VMenu>
        );
      },
    });
    await waitTime();
    // console.log(wrapper.vm.$refs.testPop);
    wrapper.vm.$refs.testPop.onPopupVisible(true);
    wrapper.vm.$nextTick(() => {
      expect(wrapper.find('.vsubmenu1').classes('v-menu-submenu-open')).toBeTruthy();
    });
    await waitTime();
    wrapper.vm.$refs.testPop.onPopupVisible(false);
    wrapper.vm.$nextTick(() => {
      expect(wrapper.find('.vsubmenu1').classes('v-menu-submenu-open')).toBeFalsy();
    });
    // expect(wrapper.find('.vsubmenu').classes('v-menu-submenu-active')).toBeTruthy();
    // expect().toBe('vsubmenu');
  });

  it('compute', async () => {
    const wrapper = mount({
      data() {
        return {
          rootMode: 'horizontal',
          isInlineMode: false,
          disabled: false,
          ancestorSubMenuNames: [],
          eventName: '',
          hasTitleAttr: false,
          trigger: ['hover'],
        };
      },
      render() {
        const {
          isInlineMode, disabled, hasTitleAttr, trigger
        } = this;
        return (
          <VMenu disabled={disabled} isInlineMode={isInlineMode}>
            <VSubMenu
              ref="testCompute"
              hasTitleAttr={hasTitleAttr}
              disabled={disabled}
              trigger={trigger}
              name="vsubmenu1"
              title="vsubmenu1"
              class="vsubmenu1"
            >
              <template slot="title">测试4</template>
              <VSubMenu ref="dddd" name="vsubmenu2" title="vsubmenu2" class="vsubmenu2">
                <template slot="title">测试5</template>
              </VSubMenu>
            </VSubMenu>
          </VMenu>
        );
      },
    });
    // wrapper.vm.rootMode = 'vertical';
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$refs.testCompute.rootSubMenu.name).toBe('vsubmenu1');
    await waitTime();
    wrapper.vm.disabled = true;
    wrapper.vm.$nextTick(() => {
      // console.log(wrapper.vm.$refs.testCompute.actions);
      expect(wrapper.vm.$refs.testCompute.actions.length).toBe(0);
    });

    // expect(wrapper.find('.vsubmenu').classes('v-menu-submenu-active')).toBeTruthy();
    // expect().toBe('vsubmenu');
  });

  it('mode', async () => {
    const wrapper = mount({
      data() {
        return {
          mode: 'inline',
        };
      },
      render() {
        const { mode } = this;
        return (
          <VMenu mode={mode}>
            <VSubMenu name="vsubmenu1" title="vsubmenu1" class="vsubmenu1">
              <template slot="title">测试4</template>
              <VSubMenu ref="testMode" name="vsubmenu2" title="vsubmenu2" class="vsubmenu2">
                <template slot="title">测试5</template>
              </VSubMenu>
            </VSubMenu>
          </VMenu>
        );
      },
    });
    // wrapper.vm.rootMode = 'vertical';
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$refs.testMode.mode).toBe('inline');
    wrapper.vm.mode = 'horizontal';
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.$refs.testMode.mode).toBe('vertical');
    });

    // expect(wrapper.find('.vsubmenu').classes('v-menu-submenu-active')).toBeTruthy();
    // expect().toBe('vsubmenu');
  });
});
