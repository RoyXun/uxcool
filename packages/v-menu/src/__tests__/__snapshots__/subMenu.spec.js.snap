// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subMenu mouseenter , mouseleave 1`] = `
<ul role="menu" class="v-menu v-menu-root v-menu-light v-menu-horizontal">
  <li class="v-menu-submenu v-menu-submenu-horizontal vsubmenu4">
    <div title="" class="v-menu-submenu-title">测试4</div>
  </li>
</ul>
`;
